File: BeeCrowd/SurnameIsNotEasy.java
// 3358 - Sobrenome não é fácil (05/02/2025)

import java.util.Scanner;

public class SurnameIsNotEasy {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int quantidade = sc.nextInt();
        String sobrenomes[] = new String[quantidade];
        for(int i = 0; i < quantidade; i++){
            sobrenomes[i] = sc.next();
        }
        
        char caracterAtual;
        String vogais = "AEIOUaeiou";

        for(int i = 0; i < quantidade; i++){
            int contador = 0;
            boolean ehFacil = true;

            for(int j = 0; j < sobrenomes[i].length(); j++){
                caracterAtual = sobrenomes[i].charAt(j);
                if(!vogais.contains(String.valueOf(caracterAtual))){
                    contador++;
                    if(contador > 2){
                        System.out.println(sobrenomes[i] + " nao eh facil");
                        ehFacil = false;
                        break;
                    }
                } else {
                    contador = 0;
                }
            }
            if(ehFacil){
                System.out.println(sobrenomes[i] + " eh facil");
            }
        }
        sc.close();
    }
}

File: BeeCrowd/SurnameIsNotEasy.c
// 3358 - Sobrenome não é fácil (05/02/2025)

#include <stdio.h>
#include <string.h>
#include <stdbool.h> 

int main(){
    int quantidade;
    scanf("%d", &quantidade);
    getchar();

    char sobrenomes[quantidade][100];
    for(int i = 0; i < quantidade; i++){
        fgets(sobrenomes[i], 100, stdin);
        sobrenomes[i][strcspn(sobrenomes[i], "\n")] = 0;
    }

    char caracterAtual;

    for(int i = 0; i < quantidade; i++){
        int contador = 0;
        bool ehFacil = true;

        for(int j = 0; j < strlen(sobrenomes[i]); j++){
            caracterAtual = sobrenomes[i][j];

            if(!strchr("AEIOUaeiou", caracterAtual)){
                contador++;
                if(contador > 2){
                    printf("%s nao eh facil\n", sobrenomes[i]);
                    ehFacil = false;
                    break;
                } 
            } else {
                contador = 0;
            }
          }
        
        if (ehFacil) {
            printf("%s eh facil\n", sobrenomes[i]);
        }
        }

    return 0;
}